src/firebolt/async_db/connection.py

```diff
@@ -222,6 +222,16 @@ async def connect_inner(
                 account_name=account_name,
                 api_endpoint=api_endpoint,
             )
+        elif account_name:
+            # We need to manually validate account_name since it's not validated
+            # in neither of above cases
```

Comment:
Should it be `if` instead? So if above cases pass it still validates the account name?

Reply:
Account name gets validated in those cases since it's used to get/resolve engine url.
I'll extend a comment explaining that